src/jobservice/job/impl/purge/purge.go

```diff
@@ -0,0 +1,116 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package purge
+
+import (
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/audit"
+	"os"
+	"strings"
+)
+
+// Job defines the purge job
+type Job struct {
+	retentionHour     int
+	includeOperations []string
+	dryRun            bool
+	auditMgr          audit.Manager
+}
+
+// MaxFails is implementation of same method in Interface.
+func (j *Job) MaxFails() uint {
+	return 1
+}
+
+// MaxCurrency is implementation of same method in Interface.
+func (j *Job) MaxCurrency() uint {
+	return 1
+}
+
+// ShouldRetry ...
+func (j *Job) ShouldRetry() bool {
```

Comment:
I am not sure that, retry here means that overwriting the last exeution or just creating a new one. This is important since if overwrite, we won't have the failure log for debugging?

Reply:
change to false